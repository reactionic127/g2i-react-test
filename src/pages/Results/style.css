.results-container {
  width: 100%;
  text-align: center;
}
.button-wrapper {
  width: 100%;
  text-align: center;
}
