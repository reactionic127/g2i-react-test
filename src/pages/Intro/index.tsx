import { Link, RouteComponentProps } from "@reach/router";

import "./style.css";

// eslint-disable-next-line @typescript-eslint/no-unused-vars
export const IntroPage = (props: RouteComponentProps) => (
  <div className="intro-container">
    <h1>Welcome to the Trivia Challenge</h1>
    <h2>You will be presented with 10 True or False questions.</h2>
    <h2>Can you score 100%?</h2>
    <Link to="/questions">BEGIN</Link>
  </div>
);
