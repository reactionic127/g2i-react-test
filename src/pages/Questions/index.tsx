import { navigate, RouteComponentProps } from "@reach/router";
import { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { QuestionCard } from "../../components/Card";
import {
  GET_QUESTIONS_REQUEST,
  POST_ANSWER_REQUEST,
} from "../../redux/constants";
import getQuestionState from "../../redux/selectors";

import "./style.css";

// eslint-disable-next-line @typescript-eslint/no-unused-vars
export const QuestionsPage = (props: RouteComponentProps) => {
  const dispatch = useDispatch();
  const { questions } = useSelector(getQuestionState);
  const [currentIndex, setCurrentIndex] = useState(0);

  useEffect(() => {
    dispatch({ type: GET_QUESTIONS_REQUEST });
  }, []);

  const handleClick = (answer: string) => {
    dispatch({ type: POST_ANSWER_REQUEST, payload: { answer } });
    setCurrentIndex(currentIndex + 1);
    if (currentIndex === questions.length - 1) {
      navigate("/results");
    }
  };

  return (
    <div className="questions-container">
      {questions[currentIndex] && (
        <>
          <h1>{questions[currentIndex].category}</h1>
          <QuestionCard
            title={questions[currentIndex].question}
            handleClick={handleClick}
          />
          <span className="question-index">
            {currentIndex + 1} / {questions.length}
          </span>
        </>
      )}
    </div>
  );
};
