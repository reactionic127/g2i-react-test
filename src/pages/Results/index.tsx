import { navigate, RouteComponentProps } from "@reach/router";
import { Button, Typography } from "antd";
import { useDispatch, useSelector } from "react-redux";
import { ResultItem } from "@components/ResultItem";
import { CLEAR_DATA_REQUEST } from "@redux/constants";
import getQuestionState from "@redux/selectors";
import { Question } from "@types";

import "./style.css";

// eslint-disable-next-line @typescript-eslint/no-unused-vars
export const ResultsPage = (props: RouteComponentProps) => {
  const dispatch = useDispatch();
  const { questions, answers } = useSelector(getQuestionState);

  const getResults = () => {
    let count = 0;
    questions.forEach((qs: Question, index: number) => {
      if (qs.correct_answer === answers[index]) {
        count++;
      }
    });
    return count;
  };

  const replay = () => {
    dispatch({ type: CLEAR_DATA_REQUEST });
    navigate("/");
  };

  return (
    <div className="results-container">
      {questions && questions.length > 0 && (
        <>
          <Typography.Title level={2}>
            You scored {getResults()} / {questions.length}
          </Typography.Title>
          {questions.map((qs: Question, index: number) => (
            <ResultItem
              key={qs.question}
              question={qs}
              answer={answers[index]}
            />
          ))}
        </>
      )}
      <div className="button-wrapper">
        <Button type="primary" onClick={replay}>
          PLAY AGAIN
        </Button>
      </div>
    </div>
  );
};
