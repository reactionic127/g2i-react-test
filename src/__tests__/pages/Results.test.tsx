import { shallow, ShallowWrapper } from "enzyme";
import { ResultsPage } from "../../pages/Results";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  useSelector: () => ({ questions: [] }),
  useDispatch: () => mockDispatch,
}));

describe("ResultsPage", () => {
  describe("rendering", () => {
    let wrapper: ShallowWrapper;
    beforeEach(() => {
      wrapper = shallow(<ResultsPage />);
    });

    it("renders the component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
