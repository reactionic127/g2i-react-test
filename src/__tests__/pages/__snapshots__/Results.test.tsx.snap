// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPage rendering renders the component 1`] = `
<div
  className="results-container"
>
  <div
    className="button-wrapper"
  >
    <Button
      onClick={[Function]}
      type="primary"
    >
      PLAY AGAIN
    </Button>
  </div>
</div>
`;
